dbfit-java/core/src/main/java/dbfit/api/AbstractDbEnvironment.java

```diff
@@ -126,14 +127,26 @@ public DdlStatementExecution createDdlStatementExecution(String ddl)
         return new DdlStatementExecution(getConnection().createStatement(), ddl);
     }
 
+    /**
+     * Create a procedure statement execution object for the given prepared statement
+     */
+    protected StatementExecution createStatementExecution(PreparedStatement statement) {
+        return new StatementExecution(statement);
+    }
+
     @Override
-    public StatementExecution createStatementExecution(PreparedStatement statement, boolean clearParameters) {
-        return new StatementExecution(statement, clearParameters);
+    public final StatementExecution createStatementExecution(String commandText) throws SQLException {
+        return createStatementExecution(getConnection().prepareStatement(commandText));
+    }
+
+    protected StatementExecution createCallExecution(PreparedStatement statement, boolean isFunction)
```

Comment:
> Do we want/need to retain methods that take a PreparedStatement or can we replace all of them with methods that take SQL text?

`createStatementExecution(PreparedStatement)` is currently used by 2 other methods. If e.g. we want to customize something - similarly like we did for Informix: it's good to minimize the number of places we need to touch. (And it's hard to get from-query-text-only version of the bound symbols method).